engine/src/main/java/org/hibernate/validator/constraints/ISBN.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.constraints;
+
+import static java.lang.annotation.ElementType.ANNOTATION_TYPE;
+import static java.lang.annotation.ElementType.CONSTRUCTOR;
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.ElementType.TYPE_USE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import javax.validation.Constraint;
+import javax.validation.Payload;
+
+import org.hibernate.validator.constraints.ISBN.List;
+
+/**
+ * Checks that the annotated character sequence is a valid
+ * <a href="https://en.wikipedia.org/wiki/International_Standard_Book_Number">ISBN</a>.
+ * The length of the number and the check digit are both verified.
+ * <p>
+ * The supported type is {@code CharSequence}. {@code null} is considered valid.
+ * <p>
+ * During validation all non ISBN characters are ignored. All digits and 'X' are considered
+ * to be valid ISBN characters. This is useful when validating ISBN with dashes separating
+ * parts of the number (ex. {@code 978-161-729-045-9}).
+ *
+ * @author Marko Bekhta
```

Comment:
@marko-bekhta Can you add an `@since` tag to this and the other new API-level types of this change?

Reply:
Ah sure, I think we forgot this as it wasn't supposed to go in.

Pushed just now.